api/openapi/specs/core_service.yaml

```diff
@@ -0,0 +1,331 @@
+        "200":
+          description: Successful request
+          content:
+            application/json:
+              schema:
+                $ref: "./api_types.yaml#/components/schemas/ServiceInfo"
+
+  /enclaves/{enclave_identifier}/services/history:
+    get:
+      description: Returns information about all existing & historical services
+      parameters:
+        - $ref: './api_types.yaml#/components/parameters/enclave_identifier'
+      responses:
+        default:
+          $ref: '#/components/responses/NotOk'
+        "200":
+          description: Successful request
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "./api_types.yaml#/components/schemas/ServiceIdentifiers"
+
+  /enclaves/{enclave_identifier}/services:
+    get:
+      parameters:
+        - $ref: './api_types.yaml#/components/parameters/enclave_identifier'
+        - in: query
+          name: services
+          schema:
+            type: array
+            items:
+              type: string
+          description: Select services to get information
+      responses:
+        default:
+          $ref: '#/components/responses/NotOk'
+        "200":
+          description: Successful request
+          content:
+            application/json:
+              schema:
+                type: object
+                additionalProperties:
+                  $ref: "./api_types.yaml#/components/schemas/ServiceInfo"
+
+  /enclaves/{enclave_identifier}/services/{service_identifier}/command:
+    post:
+      description: Executes the given command inside a running container
+      parameters:
+        - $ref: './api_types.yaml#/components/parameters/enclave_identifier'
+        - $ref: './api_types.yaml#/components/parameters/service_identifier'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "./api_types.yaml#/components/schemas/ExecCommand"
+        description: Exec Command
+        required: true
+      responses:
+        default:
+          $ref: '#/components/responses/NotOk'
+        "200":
+          description: Successful request
+          content:
+            application/json:
+              schema:
+                $ref: "./api_types.yaml#/components/schemas/ExecCommandResult"
+
+  /enclaves/{enclave_identifier}/services/{service_identifier}/endpoints/{port_number}/availability:
+    get:
+      description: Block until the given HTTP endpoint returns available, calling it through a HTTP request
+      parameters:
+        - $ref: './api_types.yaml#/components/parameters/enclave_identifier'
+        - $ref: './api_types.yaml#/components/parameters/service_identifier'
+        - $ref: './api_types.yaml#/components/parameters/port_number'
+        - $ref: './api_types.yaml#/components/parameters/http_method'
+        - $ref: './api_types.yaml#/components/parameters/path'
```

Comment:
some of these parameters are query params right? should they be specified here?

Reply:
Yes, they are specified in the `in:` field ([example](https://github.com/kurtosis-tech/kurtosis/blob/f3cb0c0d363bed2402603420b7e120d146980482/api/openapi/specs/kurtosis_api.yaml#L665))